### DISCUSSION TITLE
Support pasting rich text link on description editor

### DISCUSSION THREAD

**Discussion by r7kamura** [community/community#16206] [Projects and Issues]
I would like to see the issue and comment editors properly convert rich text links to Markdown format links when pasted ...or at least prevent oddly formatted strings from being pasted.

## Example

I created a simple JavaScript to copy a link in both HTML and plain text format:

```javascript
function copy({ html, plain }) {
  const listener = (event) => {
    event.clipboardData.setData("text/html", html);
    event.clipboardData.setData("text/plain", plain);
    event.preventDefault();
  };
  document.addEventListener("copy", listener);
  document.execCommand("copy");
  document.removeEventListener("copy", listener);
}

copy({
  html: '<a href="http://example.com/">foo</a>',
  plain: "foo bar"
})
```

We can try this by pasting the above code to devtools console.

When we try to paste the copied data into the editor on GitHub Issues, the following string is pasted:

```
[foo](http://example.com/) bar
```

...but, isn't this a little odd?

Ideally, I would like the following string pasted:

```
[foo bar](http://example.com/)
```

Or, less ideally, I would like to see the following plain text pasted:

```
foo bar
```

**Comment by bartlettroscoe** [comment]
NOTE: GitLab has this feature and I use it all of the time to copy and paste formatted text into an GitLab issue comment.  It is a big deal.

NOTE: Windows 11 PowerToys has an "Advanced Paste" feature where you can copy and paste formatted text as markdown.  You just have to assign a key combo for it.  I just used "Ctrl+Win+Alt+V" for this.  Here is an example:

[vscode#254473](https://github.com/microsoft/vscode/issues/254473): Add scripting interface/batch mode for Copilot Agent Mode?
